osquery/database/plugins/ephemeral.cpp

```diff
@@ -16,10 +18,36 @@ Status EphemeralDatabasePlugin::get(const std::string& domain,
                                     const std::string& key,
                                     std::string& value) const {
   if (db_.count(domain) > 0 && db_.at(domain).count(key) > 0) {
-    value = db_.at(domain).at(key);
+    try {
+      value = boost::get<std::string>(db_.at(domain).at(key));
+    } catch (const boost::bad_get& e) {
+      LOG(WARNING) << "Type error getting string value for (domain,key) : ("
+                   << key << "," << domain << ") " << e.what();
+      return Status(
+          1,
+          "EphemeralDatabasePlugin::get was requested incorrect type(string)");
+    }
+    return Status(0);
+  } else {
+    return Status(1, "Key or domain does not exist");
+  }
+}
+
+Status EphemeralDatabasePlugin::get(const std::string& domain,
+                                    const std::string& key,
+                                    int& value) const {
+  if (db_.count(domain) > 0 && db_.at(domain).count(key) > 0) {
+    try {
+      value = boost::get<int>(db_.at(domain).at(key));
```

Comment:
This is triple search for `domain` in map and double search for `key`. I bet you can use just one for each of it :)

Reply:
Nice catch, I also noticed that, however, don't want to overcomplicate this pull request ( this code was like that before). Will definitely change it in the next pull request.

Reply:
Added now, it also helps for better error message (y)